### ISSUE TITLE
Postman v7.23.0 snap does not open in Fedora 31 or Fedora 32

### ISSUE THREAD

**Issue by DavidRedstone** [a85/POSTMan-Chrome-Extension#8425]
Description:
Sorry if this is a dupe. I couldn't find it listed. Postman v7.23.0 snap does not open in Fedora 31 or Fedora 32.

Steps to reproduce:
Install Postman via snap store or via sudo snap install postman
Open postman from the programs list or cli

Expected behavior:
Postman should open, but nothing happens. 
If opened via cli, there is a message about cgroups v2

App info:
Latest postman snap app - v7.23.0 

Additional Info:
Every other snap app I have, works as expected (Spotify, VLC, Discord, Remmina, Anki, VS Code, Sublime Text, Slack, Signal...) Only Postman fails to open. I upgraded from Fedora 31 to Fedora 32 and everything still works, except Postman.

For the time being I am using the normal download from postman's website. Unfortunately that one doesn't install itself as an app in the programs list, so I have to open it from the file, but at least it opens.

**Comment by harryi3t** [comment]
@DavidRedstone Did you do a fresh install of Postman snap app or were you using it for a while and it broke with the latest update? If the latter is the case, do you remember that last version that was working fine on this system?

**Comment by DavidRedstone** [comment]
Hi Harry,

Thanks for the quick reply.

At work I use Postman on F31 (and prev, F30) on a desktop PC, but from the normal download. At home, I have tried the snap on an HP laptop with F32 (and prev F31), and I have never gotten the snap working. The only hint is the cgroups v2 warning.

I also have a media PC with Ubuntu 19.10... installing Postman snap to check... yes it works. So far everything points to interaction with cgroups v2, as being the issue. 

I know I can set cgroups back to v1 with grubby, but shouldn't have to. 

`sudo dnf install grubby`
`sudo grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"`

Fedora defaulted to cgroups v2 last year. Fwiw, other notable software that isn't cgroups v2 compatible, is Docker.

I hope this helps.

**Comment by DavidRedstone** [comment]
More info (from Fedora 32)...

Install Postman snap:
`[dave@silverbeast ~]$ sudo snap install postman
Warning: /var/lib/snapd/snap/bin was not found in your $PATH. If you've not restarted your session
         since you installed snapd, try doing that. Please see https://forum.snapcraft.io/t/9469
         for more details.

postman 7.24.0 from Postman, Inc. (postman-inc✓) installed
`

Run Postman (from CLI, to show output -- Postman does not open):
`[dave@silverbeast ~]$ postman
WARNING: cgroup v2 is not fully supported yet, proceeding with partial confinement
[dave@silverbeast ~]$ `

Try with full path:
`[dave@silverbeast ~]$ /var/lib/snapd/snap/bin/postman
WARNING: cgroup v2 is not fully supported yet, proceeding with partial confinement
[dave@silverbeast ~]$ 
`

Run Postman from CLI, with sudo (Postman not found):
`[dave@silverbeast ~]$ sudo postman
sudo: postman: command not found
`

Run Postman from CLI, with sudo and full path (Postman opens, yay!)
`[dave@silverbeast ~]$ sudo /var/lib/snapd/snap/bin/postman
WARNING: cgroup v2 is not fully supported yet, proceeding with partial confinement
Gtk-Message: Failed to load module "canberra-gtk-module"
Gtk-Message: Failed to load module "pk-gtk-module"
Gtk-Message: Failed to load module "canberra-gtk-module"
Gtk-Message: Failed to load module "pk-gtk-module"
Not disabling GPU
ProtocolHandler~setDefaultProtocolClient: Command failed: xdg-mime default Postman.desktop x-scheme-handler/postman
/bin/sh: 1: xdg-mime: not found

Failed to open connection to "session" message bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
1589416297206 main info "Booting Postman 7.24.0, linux-5.6.11-300.fc32.x86_64 on x64"
1589416297207 main info "EventBus~initialize - Success"
1589416297210 main warn "CloudProxyManager~init: Error while reading proxy configuration from user data - ",{"errno":-2,"code":"ENOENT","syscall":"open","path":"/root/snap/postman/108/.config/Postman/Postman_Config/proxy","name":"Error","message":"ENOENT: no such file or directory, open '/root/snap/postman/108/.config/Postman/Postman_Config/proxy'","stack":"Error: ENOENT: no such file or directory, open '/root/snap/postman/108/.config/Postman/Postman_Config/proxy'"}
1589416297210 main info "CloudProxyManager~init - Success"
1589416297223 main info "UpdateHandler~init - Success"
1589416297224 main info "RuntimeExecutionService~initialized: Success"
1589416297737 main info "LinuxAutoUpdater~Cleanup - Initial cleanup successful"
1589416297785 main warn "main~populateInstallationId: Failed to get installationId from appSettings but with no error"
1589416298221 main info "Bootstrap-models~bootstrap - Success"
1589416299329 main info "window-manager~newSharedWindow: Shell loaded"
1589416304468 main info "Main~AppEvents - Received booted event for process shared"
INTERCEPTOR CONNECTIVITY: Connecting to Interceptor Bridge
InterceptorBridge: Trying to connect Native App IPC~Interceptor Bridge
1589416307801 main info "Main~AppEvents - Received booted event for process requester"
IPCClient ~ WindowHandler: New event is received { type: 'window-opened' }
1589416334218 main info "WindowManager~closeHandler - Closed Window (id: 4 )"
IPCClient ~ WindowHandler: New event is received { type: 'window-closed', windowId: 4 }
1589416334232 main info "Quitting app"
[dave@silverbeast ~]$ 
`

Long story short, on Fedora 32 (and 31) the latest snap only opens when run from CLI with sudo and the full path. 

Then there is another problem. I can't sign in with Google, because the snap can't open a browser window. This is not a problem with other snaps, that don't need sudo to run.

I hope this helps :)
Dave

**Comment by twitharshil** [comment]
@DavidRedstone can you update the app to the latest version [As of now the latest version is 7.29.1] and let us know if you are still facing the issue?
After installing the postman app using snap, you can start the application using the following command on CLI 
`snap run postman`

**Comment by DavidRedstone** [comment]
Thanks for the reply :+1: 

I've since switched to Ubuntu 20.04 (multiboot with Fedora 32 and Windows 10), but next time I boot into Fedora 32 I will try the Postman Snap again, and see how it goes. 

The latest Postman Snap works well on Ubuntu 20.04:
`dave@Silverbeast:~$ snap list | grep post
postman              7.29.1                      119   latest/stable    postman-inc*  -
`

**Comment by DavidRedstone** [comment]
@twitharshil I just booted into Fedora 32 to test the Postman snap for you. Turns out I had fully uninstalled snap, so this was stating fresh...

```
sudo dnf install snapd -y
sudo ln -s /var/lib/snapd/snap /snap
sudo snap install postman -y
snap run postman
```
Good news: 
✓    Postman opened, and fairly quickly at that, 
✓    'Sign in with Google' opened my browser and I successfully signed in
✓    My collections synced across

Whatever you changed, worked, so thank you. :+1: :1st_place_medal: 

Cheers
Dave

**Comment by twitharshil** [comment]
Thanks for the confirmation @DavidRedstone, Glad that the issue is fixed :). Marking this issue closed 👍
